### PR TITLE
Sam/15 16 nix package, build and test of 15.8 and 16.3

### PR DESCRIPTION
**Pull request by samrose**
## What kind of change does this PR introduce?

This makes all the necessary changes to package, build and test in the nix/pg_regress context our psql bundles with 15.8 and 16.3

We attempt to change no versions, and as much as possible to keep 1 pg_regress test suite, and one set of extensions and wrappers. 

There will be a follow up PR to address the packer/ansible/gh action work for integration tests and releasing images.

It includes the changes to packer/ansible/gh actions to allow for a matrix build of pg versions on release.

## Significant changes

This PR moves the source of truth for data about postgres major and semver back to ansible/vars.yml, which are then read at time of release in the ami-release gh action. The _**common-vars.nix.yml file is no longer under source control**_, but instead is generated duing the ci run for ami release or testinfra runs. The version is now controlled in this allows us to avoid making too many changes to our release and test tooling for AMI)

To bump the versions of postgres it is proposed to update these values that are now in `anisble/vars.yml`

These versions cover both release and testinfra

```yml
postgres_major:
  - "15"
  - "16"

# Full version strings for each major version
postgres_release:
  postgres15: "15.8.1.003-staging-5"
  postgres16: "16.3.1.000-staging-5"
```

### CODE REVIEW

**Review by olirice — APPROVED** [review]

File: migrations/Dockerfile.dbmate [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+RUN echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" | tee /etc/apt/sources.list.d/pgdg.list
+RUN apt-get update && apt-get install -y postgresql-client-%VERSION%
+
+ENV PATH="/usr/lib/postgresql/%VERSION%/bin:${PATH}"
+
+RUN dbmate --version
+
+ENTRYPOINT ["dbmate"]
```
Comment (pcnc) [inline migrations/Dockerfile.dbmate]: New line here

File: .github/workflows/testinfra-nix.yml [hunk 2]
```diff
@@ -91,4 +120,4 @@ jobs:
           
           # Deregister AMIs
           deregister_ami_by_name "$STAGE1_AMI_NAME"
-          deregister_ami_by_name "$STAGE2_AMI_NAME"
```
Comment (pcnc) [inline .github/workflows/testinfra-nix.yml:94]: New line here

**Comment by pcnc** [comment]
How do these workflow changes impact the workflows which are tightly bound to the `common-nix-vars.pkr.hcl` file? Should we parse `ansible/vars.yml` outputs to ensure they continue working as intended?

i.e.

https://github.com/supabase/postgres/blob/develop/.github/workflows/publish-nix-pgupgrade-scripts.yml
https://github.com/supabase/postgres/blob/develop/.github/workflows/publish-nix-pgupgrade-bin-flake-version.yml
https://github.com/supabase/postgres/blob/develop/.github/workflows/test.yml

---

The current workflow generates a git tag for every AMI release - do these changes maintain this behaviour?

**Comment by samrose** [comment]
> How do these workflow changes impact the workflows which are tightly bound to the `common-nix-vars.pkr.hcl` file? Should we parse `ansible/vars.yml` outputs to ensure they continue working as intended?
> 
> i.e.
> 
> https://github.com/supabase/postgres/blob/develop/.github/workflows/publish-nix-pgupgrade-scripts.yml https://github.com/supabase/postgres/blob/develop/.github/workflows/publish-nix-pgupgrade-bin-flake-version.yml https://github.com/supabase/postgres/blob/develop/.github/workflows/test.yml
> 
> The current workflow generates a git tag for every AMI release - do these changes maintain this behaviour?

Good point thank you!

I did adjust  https://github.com/supabase/postgres/blob/develop/.github/workflows/test.yml but I did not address 


https://github.com/supabase/postgres/blob/develop/.github/workflows/publish-nix-pgupgrade-scripts.yml https://github.com/supabase/postgres/blob/develop/.github/workflows/publish-nix-pgupgrade-bin-flake-version.yml 

...so I will take a look at those today

The general approach I was taking was to generate the hcl file at build time, and move the values to vars.yml

**Comment by samrose** [comment]
> The current workflow generates a git tag for every AMI release - do these changes maintain this behaviour?

Yes they do as demonstrated when I ran a staging release they are visible at https://github.com/supabase/postgres/releases  (those with *-staging-*)

**Comment by samrose** [comment]
@pcnc I believe these changes will handle the pg upgrade actions. Although I am not sure if it is enough to go across major versions, or if more changes are needed to achieve sam 15.8 or 15.6 -> 16.3 nix -> nix pg_upgrade process with these changes so please let me know https://github.com/supabase/postgres/pull/1266/commits/3454b624004a1f2a3f8823941768af3412357c87

**Comment by samrose** [comment]
Going to address the docker work in a soon to be completed follow up PR to try to keep this work moving along, and break up into smaller chunks

**Comment by pcnc** [comment]
@samrose that works great, thanks!

Only nit I'd have is that we'd want to preserve the overrides (the `if [[ "${{ inputs.postgresVersion }}" != "" ]];` blocks) in order to deliver script/upgrade flake hash hotfixes for previous releases if required 😬

**Comment by samrose** [comment]
@pcnc ok https://github.com/supabase/postgres/pull/1266/commits/312e4d7922926870d7f5fa18aa1baa22116ce01f should take care of that if it looks good can you approve and we'll get it merged. Thank you!

**Comment by pcnc** [comment]
Looks good, thanks! Approved.

**Review by pcnc — APPROVED** [review]

File: migrations/Dockerfile.dbmate (hunk 1 above)
Comment (pcnc) [reply migrations/Dockerfile.dbmate]: These were older comments which were not submitted for some reason.

File: .github/workflows/testinfra-nix.yml (hunk 2 above)
Comment (pcnc) [reply .github/workflows/testinfra-nix.yml:94]: These were older comments which were not submitted for some reason.

Comment (samrose) [reply .github/workflows/testinfra-nix.yml:94]: Resolved thanks

**Pull request merged**